### PR TITLE
New communities redesign 

### LINKED ISSUE #13177: Communities redesign
Implement the new communities designs https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?node-id=0%3A1

### PR DESCRIPTION
**Pull request by jo-mut**
fixes #13177

### Summary
This is an implementation of communities redesign as required in the design input

### Testing notes
Navigate to profile
--> Advanced
--> Toggle New UI

### CODE REVIEW

**Comment by du82** [comment]
I really like the 3D style, but most people have 5 fingers, and Status doesn't support Bitcoin.

Screenshot_20220707-154910_Brave

**Comment by du82** [comment]
Gating icons are covering text in the small card view

Screenshot_20220722-171003_Status PR

**Comment by du82** [comment]
Two issues: Buttons clip the status bar, and the cards clip the scrollview on the right
Screenshot_20220807-214213_Status PR

File: src/quo/components/separator.cljs
Comment (flexsurfer) [inline src/quo/components/separator.cljs]: it should be in quo2 then

File: src/status_im/i18n/i18n.cljs
Comment (flexsurfer) [inline src/status_im/i18n/i18n.cljs]: how its related to i18n ?

File: src/status_im/subs.cljs
Comment (flexsurfer) [inline src/status_im/subs.cljs]: could you elaborate on why bottom sheet redesing is needed ?

File: src/status_im/bottom_sheet/core.cljs
Comment (flexsurfer) [inline src/status_im/bottom_sheet/core.cljs]: i guess you could keep old bottom sheet , just use `new-ui-eneabled` in bottom sheet itself if you need to change UI of bottom sheet

File: src/status_im/react_native/resources.cljs
Comment (flexsurfer) [inline src/status_im/react_native/resources.cljs]: it could be just `:community-cover`

File: src/quo2/components/separator.cljs
Comment (flexsurfer) [inline src/quo2/components/separator.cljs:1]: oh i see you have it in quo2 , probably you just need to delete old one in quo

**Review by flexsurfer — COMMENTED** [review]
it defenetly will take time to review, i see duplicated changes from other PRs, probably we should wait those PRs merged first and then continue review

**Comment by jo-mut** [comment]
> it defenetly will take time to review, i see duplicated changes from other PRs, probably we should wait those PRs merged first and then continue review

Yeah. initially I had worked on the whole screen but switched to implementing the components which is the reason for the duplication. Once I merge the components this PR will gradually reduce

File: src/status_im/i18n/i18n.cljs
Comment (jo-mut) [reply src/status_im/i18n/i18n.cljs]: This one is not at all related. I place this in the right ns

**Comment by jo-mut** [comment]
I will just tag this with WIP so first I can focus on the components which are part of this PR

**Comment by jo-mut** [comment]
hi @qoqobolo 

I would like to merge communities redesign and skip the design review process.

**Comment by qoqobolo** [comment]
Hey @jo-mut, not sure I'm the right person for approving this :)

There are several points:
1. Was this agreed upon with the design team?
2. As agreed, every redesign PR should go through e2e
3. Redesign PRs don't go through manual testing, so we are not currently involved in the approval/rejection merge process
4. I can see that this PR is not still approved by @flexsurfer after review

Could you clarify why are you going to skip all these steps, please?

**Comment by jo-mut** [comment]
the design team is okay with that. Once its approved @flexsurfer and the QA is okay thats would be fine

**Comment by qoqobolo** [comment]
> the design team is okay with that. Once its approved @flexsurfer and the QA is okay thats would be fine

Okay, then put the `Request-manual-qa` label (if it needs to be tested manually) and move PR to the `E2E Tests` column after the reviewer's approval. It will be tested in the queue order.
And make sure that PR is rebased, please.
Thank you

**Comment by jo-mut** [comment]
cool thanks @qoqobolo

**Comment by du82** [comment]
The buttons and text doesn't line up with the rest of the app's buttons

https://user-images.githubusercontent.com/65834069/185960876-5dd4dc0b-59c3-4b92-b8b2-eb1b20a180c0.mp4

**Comment by du82** [comment]
Community cards are also cut off on the right side

Screenshot_2022-08-22-10-38-18-42_65ad86f86fd2321a85272a99323da10f

File: src/quo2/components/separator.cljs
Comment (flexsurfer) [reply src/quo2/components/separator.cljs:1]: still not addressed

File: src/status_im/react_native/resources.cljs
Comment (flexsurfer) [reply src/status_im/react_native/resources.cljs]: still not addressed

File: src/status_im/subs.cljs
Comment (jo-mut) [reply src/status_im/subs.cljs]: i will just use  new-ui-enabled

**Comment by jo-mut** [comment]
@flexsurfer I have addressed your feedback on this. Looks fine

File: 1
Comment (flexsurfer) [inline 1]: whats that? :)

File: src/quo/components/bottom_sheet/style.cljs
Comment (flexsurfer) [inline src/quo/components/bottom_sheet/style.cljs:29]: yeah it's not cool, but i guess its fine for now while we in transition to quo2

File: src/status_im/i18n/i18n.cljs
Comment (flexsurfer) [inline src/status_im/i18n/i18n.cljs]: could you rollback this file pls

File: src/status_im/subs.cljs
Comment (flexsurfer) [inline src/status_im/subs.cljs]: could you elaborate? cc @Parveshdhull

File: 1
Comment (Parveshdhull) [inline 1]: irrelevant file?

File: src/status_im/ui/screens/communities/sort_communities.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/sort_communities.cljs]: better to use `[<>` in such cases

**Comment by Parveshdhull** [comment]
why these two icons don't have size
image

**UPD**:
Also, same with group, and please don't delete old icons, they are still being used by old-ui

image

File: 1
Comment (jo-mut) [reply 1]: yeah very irrelevant. am removing it

**Comment by jo-mut** [comment]
> why these two icons don't have size image

they shouldnt be there actually only size 20 is needed for this

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (siddarthkay) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: not a huge thing but `case` is better than `cond` here 
Ref: https://github.com/bbatsov/clojure-style-guide#case-vs-condcondp

File: src/status_im/ui/screens/communities/styles.cljs
Comment (siddarthkay) [inline src/status_im/ui/screens/communities/styles.cljs]: opportunity to move this to colors namespace?

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: `[:<>`

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: its better to move away from defview and letsubs to `<sub` function

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: `:notifications-button`

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: `get-item-layout-js`

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: `defn`

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: i18n

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: why view wrapper ?

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: `[:<>`

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs:87]: space between functions

File: src/status_im/ui/screens/communities/community_membership_status.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/community_membership_status.cljs]: `<sub` , just use `<sub` everywhere

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: i believe i've seen this component in other screens, if its a duplication can we move them in separate ns ?

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: `<sub`

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: `[:<>]`

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: i18n

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: defn

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: defn

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: i18n

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: i18n

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: defn

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: i18n

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/communities/communities_list_redesign.cljs]: `<sub`

File: src/quo2/components/community_card_view.cljs
Comment (flexsurfer) [inline src/quo2/components/community_card_view.cljs]: better to use keywords `:gated` `:featured`

File: src/quo2/components/community_card_view.cljs
Comment (flexsurfer) [inline src/quo2/components/community_card_view.cljs]: i18n

File: src/quo2/components/community_card_view.cljs
Comment (flexsurfer) [inline src/quo2/components/community_card_view.cljs]: i18n

File: src/quo2/components/community_card_view.cljs
Comment (flexsurfer) [inline src/quo2/components/community_card_view.cljs]: why keyword ? `100%`

File: src/quo2/components/community_card_view.cljs
Comment (flexsurfer) [inline src/quo2/components/community_card_view.cljs]: could you elaborate? im not sure we should use `window-width`

File: 1
Comment (jo-mut) [reply 1]: this is a file that got created when i was ammending a commit

File: src/status_im/subs.cljs
Comment (Parveshdhull) [reply src/status_im/subs.cljs]: Looks like accidental removal, please revert.

File: src/quo2/components/community_card_view.cljs
Comment (jo-mut) [reply src/quo2/components/community_card_view.cljs]: cool

File: src/quo2/components/community_card_view.cljs
Comment (jo-mut) [reply src/quo2/components/community_card_view.cljs]: oooh actually i could use window-width

File: src/quo2/components/community_card_view.cljs
Comment (jo-mut) [reply src/quo2/components/community_card_view.cljs]: the community-card-view when used in a horizontal scroll view needs to have a width property otherwise it extend beyond the view

Comment (jo-mut) [reply src/quo2/components/community_card_view.cljs]: I avoided using pixels as it looks different than the design on different devices

Comment (Parveshdhull) [reply src/quo2/components/community_card_view.cljs]: You can try using `:flex` with padding or margin?

File: src/status_im/ui/screens/communities/communities_list_redesign.cljs
Comment (jo-mut) [reply src/status_im/ui/screens/communities/communities_list_redesign.cljs]: cool

**Comment by Parveshdhull** [comment]
hi @jo-mut, Please revert deletion of icons, (they might be still used in old-ui)?

> resources/images/icons/info@2x.png
> resources/images/icons/info@3x.png
> resources/images/icons/scan2@2x.png
> resources/images/icons/scan2@3x.png
> resources/images/icons/search2@2x.png
> resources/images/icons/search2@3x.png

And same goes for renaming, if icon already exist, either reuse that, or add new
> resources/images/icons/security24@2x.png

**Comment by jo-mut** [comment]
> hi @jo-mut, Please revert deletion of icons, (they might be still used in old-ui)?
> 
> > resources/images/icons/info@2x.png
> > resources/images/icons/info@3x.png
> > resources/images/icons/scan2@2x.png
> > resources/images/icons/scan2@3x.png
> > resources/images/icons/search2@2x.png
> > resources/images/icons/search2@3x.png
> 
> And same goes for renaming, if icon already exist, either reuse that, or add new
> 
> > resources/images/icons/security24@2x.png

these Icons are actually part of develop. It was a mistake to delete them. Thanks

File: src/quo2/components/community_card_view.cljs
Comment (jo-mut) [reply src/quo2/components/community_card_view.cljs]: looks like the `:flex` doesnt work for  list-item width in a horizontal flat-list

Comment (jo-mut) [reply src/quo2/components/community_card_view.cljs]: in this case a list-item used in a horizontal flat-list looks like should have the width property as `:flex` doesnt seem to work. This is as explained here https://snack.expo.dev/H1-wnC5HM

**Comment by jo-mut** [comment]
thankyou for your feedback. I know its a tone of work but I think I have addressed all this feedback.

**Comment by du82** [comment]
Two things:

When a user long presses or taps the community, the banner clips the card

Screenshot_2022-08-25-16-48-41-50_65ad86f86fd2321a85272a99323da10f

The color changes every time you change the card view

https://user-images.githubusercontent.com/65834069/186775407-60109cc5-82c1-4111-815a-725e35741e70.mp4

File: src/quo/components/bottom_sheet/style.cljs
Comment (Parveshdhull) [inline src/quo/components/bottom_sheet/style.cljs]: hi @jo-mut, when `new-ui?` will be false it will use `(:backdrop @colors/theme)` as `:background-color` instead of `(:ui-background @colors/theme)`. Wouldn't that will affect old ui?

File: src/quo/components/list/item.cljs
Comment (Parveshdhull) [inline src/quo/components/list/item.cljs:49]: where `:dark` is being used?

File: src/quo2/components/base_tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/base_tag.cljs:10]: `(/ size 2)` ?

File: src/quo2/components/base_tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/base_tag.cljs:25]: Thank you very much for adding comment, But it don't look like it matches parameters.

File: src/quo2/components/base_tag.cljs
Comment (Parveshdhull) [inline src/quo2/components/base_tag.cljs:31]: why `merge`?

File: src/quo2/components/community_card_view.cljs
Comment (Parveshdhull) [inline src/quo2/components/community_card_view.cljs:22]: `cond` will be more readable.

File: src/quo2/components/community_card_view.cljs
Comment (Parveshdhull) [inline src/quo2/components/community_card_view.cljs]: we should not use `>evt` or `<sub` in components

**Comment by Parveshdhull** [comment]
> hi @qoqobolo
> 
> I would like to merge communities redesign and skip the design review process.

Hi @jo-mut, Thank you very much for really awesome work.

Even if you want to skip the design review process (and design team is ok with that), wouldn't it will be easier if you create a separate PR for each component, and get them merge first.

As this PR has several standalone independent quo2 components and screens (7),
breaking them in separate PR (**one at a time**) will also help dev review process. wdyt?

**Comment by flexsurfer** [comment]
hey @Parveshdhull it was my suggestion as well for splitting this PR, but i think now we just want to merge it as is, so it won't block other devs

**Comment by jo-mut** [comment]
@Parveshdhull this PR has several components. I had created PRs for them already but the process was slow. I think its okay to merge it as one big PR and still keep working on it where implementation/code can be improved

**Comment by du82** [comment]
I'm not sure if this was intended, but a bunch of header bars, and bottom sheets have changed color in weird ways

Screenshot_2022-08-28-18-07-51-84_65ad86f86fd2321a85272a99323da10f

**Comment by jo-mut** [comment]
@du64 this was an  earlier build and I made updates to correct that

**Comment by jo-mut** [comment]
@Parveshdhull @flexsurfer is this okay to proceed with merging

**Comment by flexsurfer** [comment]
@jo-mut im ok if all my comments were addressed and resolved

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**
